data/frame_json.go

```diff
@@ -308,11 +328,40 @@ func readFrameData(iter *jsoniter.Iterator, frame *Frame) error {
 						}
 					}
 				} else {
-					iter.ReadAny() // skip nills
+					iter.ReadAny() // skip nils
+				}
+				fieldIndex++
+			}
+
+		case "nanos":
+			fieldIndex := 0
+			for iter.ReadArray() {
+				field := frame.Fields[fieldIndex]
+
+				t := iter.WhatIsNext()
+				if t == jsoniter.ArrayValue {
+					for idx := 0; iter.ReadArray(); idx++ {
+						ns := iter.ReadInt64()
+						if readValues {
+							t, ok := field.vector.ConcreteAt(idx)
+							if !ok {
+								continue
+							}
+							tWithNS := t.(time.Time).Add(time.Nanosecond * time.Duration(ns))
+							field.vector.SetConcrete(idx, tWithNS)
+							continue
+						}
+						if idx == 0 {
+							nanos[fieldIndex] = append(nanos[fieldIndex], ns)
+						}
+					}
+				} else {
+					iter.ReadAny() // skip nils
 				}
 				fieldIndex++
 			}
 
+			readNanos = true
 		default:
 			iter.ReportError("bind l2", "unexpected field: "+l2Field)
```

Comment:
This makes this a breaking change in terms of compatibility.  If a newer version creates nanos, a (backend reader) will error.

Perhaps we should consider handling differently?

Reply:
+1 -- I think this is OK now because we don't yet send JSON we are expecting to parse in plugins.

That said -- lets change the `ReportError` to either ignore, or log so we can make changes like this in the future without a difficult upgrade path